src/components/CreateDbForm.tsx

```diff
@@ -8,29 +8,46 @@ import { IoMdHand } from "react-icons/io";
 
 const CreateDbForm: React.FC = withRouter(({ history }) => {
   const [name, setName] = useState("");
+  const [isPublic, setPublic] = useState(true);
   const injector = useDependencyInjector();
   const [loading, setLoading] = useState(false);
   const [error, setError] = useState("");
   let dbProvider: MutableRefObject<DatabaseProvider> = useRef(null);
   let db: MutableRefObject<Store> = useRef(null);
 
+  // Initialises the database provider, whenever the injector changes, reinstantiate provider
+  useEffect(() => {
+    const initialiseDBProvider = async () => {
+      dbProvider.current = await injector.createDBProvider();
+    };
+
```

Comment:
If you intitialise the dbprovider here, maybe also move the teardown here?
That way, the object can leak out in case the user changes page somehow.
https://reactjs.org/docs/hooks-effect.html#effects-with-cleanup use this to call tearDown